sim/priest/starshards.go

```diff
@@ -0,0 +1,108 @@
+package priest
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/wowsims/classic/sim/core"
+)
+
+const StarshardsRanks = 7
+const StarshardsTicks = 6
+
+var StarshardsSpellId = [StarshardsRanks + 1]int32{0, 10797, 19296, 19299, 19302, 19303, 19304, 19305}
+var StarshardsTickSpellId = [StarshardsRanks + 1]int32{0, 19350, 19351, 19352, 19353, 19354, 19355, 19356}
+var StarshardsBaseDamage = [StarshardsRanks + 1]float64{0, 84, 162, 288, 414, 570, 756, 936}
+var StarshardsManaCost = [StarshardsRanks + 1]float64{0, 50, 85, 140, 190, 245, 300, 350}
+var StarshardsLevel = [StarshardsRanks + 1]int{0, 10, 18, 26, 34, 42, 50, 58}
+
+func (priest *Priest) registerStarshards() {
+	if !priest.Race.NightElf {    //not sure, if this works.. can anyone confirm? xD
+		return
+	}
+
+	priest.Starshards = make([][]*core.Spell, StarshardsRanks+1)
+
+	for rank := 1; rank <= StarshardsRanks; rank++ {
+		priest.Starshards[rank] = make([]*core.Spell, StarshardsTicks+1)
+
+		var tick int32
+		for tick = 0; tick < StarshardsTicks; tick++ {
+			config := priest.newStarshardsSpellConfig(rank, tick)
+
+			if config.RequiredLevel <= int(priest.Level) {
+				priest.Starshards[rank][tick] = priest.RegisterSpell(config)
+			}
+		}
```

Comment:
We shouldn't need to register different tick lengths of starshards since it's a DoT, not channeled. You should be able to copy how SW:P is handled instead